lib/BirthdayCalendarGenerator.php

```diff
@@ -0,0 +1,204 @@
+<?php
+
+namespace Sabre\VObject;
+
+use DateTimeInterface;
+use DateTimezone;
+use Sabre\VObject\Component\VCalendar;
+
+/**
+ * This class generates birthday calendars.
+ *
+ * @copyright Copyright (C) 2011-2015 fruux GmbH (https://fruux.com/).
+ * @author Dominik Tobschall (http://tobschall.de/)
+ * @license http://sabre.io/license/ Modified BSD License
+ */
+class BirthdayCalendarGenerator {
+
+    /**
+     * Input objects.
+     *
+     * @var array
+     */
+    protected $objects = [];
+
+    /**
+     * Reference timezone.
+     *
+     * When generating events, and we come across so-called
+     * floating times (times without a timezone), we use the reference timezone
+     * instead.
+     *
+     * This is also used for all-day events.
+     *
+     * This defaults to UTC.
+     *
+     * @var DateTimeZone
+     */
+    protected $timeZone;
+
+    /**
+     * Default year.
+     * Used for dates without a year.
+     *
+     * @var int
+     */
+    protected $defaultYear = 1900;
```

Comment:
I don't know if I came up with 1900 here... but I would actually suggest putting this a lot further in the future. Putting it on 1900 means that usually 115 recurrence instances will be created to calculate birthdays, and this will not be very efficient. I'd set it to 2010 I think.